### PR TITLE
Refactored to remove incorrect dependencies.

### PR DESCRIPTION
**Pull request by sonnykt**
JIRA: https://digital-engagement.atlassian.net/browse/SDPA-2735

### CODE REVIEW

**Comment by anthony-malkoun** [comment]
Hey @sonnykt I think some of these changes are already release... can you pull develop in and push your branch?

File: src/TideApiHelper.php [hunk 1]
```diff
@@ -181,4 +181,173 @@ public function getFrontPagePath() {
     return $this->frontPage;
   }
 
+  /**
```
Comment (anthony-malkoun) [inline src/TideApiHelper.php]: :+1: I like all these refactors and extractions.

File: tide_api.info.yml [hunk 2]
```diff
@@ -4,8 +4,8 @@ type: module
 package: Tide
 core: 8.x
 dependencies:
-  - drupal:jsonapi (>=8.x-2.0)
-  - drupal:jsonapi_extras (>=8.x-3.3)
+  - drupal:jsonapi (>=8.x-2.4)
```
Comment (anthony-malkoun) [inline tide_api.info.yml]: I didn't actually know you could do this. What actual effect does it have? Will the module fail to enable if the version in composer.json is not consistent?

File: src/Controller/TideApiController.php [hunk 3]
```diff
@@ -101,47 +115,31 @@ class TideApiController extends ControllerBase {
...
-        $container->get('path.alias_manager'),
-        $container->get('entity_type.manager'),
-        $container->get('jsonapi.resource_type.repository'),
-        $container->get('event_dispatcher'),
-        $container->get('tide_api.helper'),
-        $container->get('tide_api.repository'),
-        $container->get('language_manager')
-      );
-    }
+    return new static(
```
Comment (anthony-malkoun) [inline src/Controller/TideApiController.php]: 👍

File: src/Controller/TideApiController.php [hunk 4]
```diff
@@ -154,101 +152,55 @@ public static function create(ContainerInterface $container) {
    *   JSON response.
    */
   public function getRoute(Request $request) {
-    global $base_url;
     $code = Response::HTTP_NOT_FOUND;
```
Comment (anthony-malkoun) [inline src/Controller/TideApiController.php]: This looks much better 👍

File: src/Controller/TideApiController.php [hunk 5]
```diff
@@ -272,18 +218,43 @@ public function getRoute(Request $request) {
...
-    if ($code == 404) {
-      if ($path !== '/' && $redirect = $this->redirectRepository->findMatchingWildcardRedirect($path,
-          $query, $this->languageManager->getCurrentLanguage()->getId())) {
-        $this->resolveRedirectPath($redirect, $site, $json_response, $code);
+
+    // Dispatch a GET_ROUTE event so that other modules can modify it.
+    if ($code != Response::HTTP_BAD_REQUEST) {
+      $entity = $this->routeEntity ? clone $this->routeEntity : NULL;
+      $event = new GetRouteEvent(clone $request, $json_response, $entity, $code, clone $this->cacheMetadata);
+      $this->eventDispatcher->dispatch(TideApiEvents::GET_ROUTE, $event);
```
Comment (anthony-malkoun) [inline src/Controller/TideApiController.php]: This makes sense that it is called here rather than later 👍

File: tide_api.info.yml (hunk 2 above)
Comment (sonnykt) [reply tide_api.info.yml]: Drupal won't enable the module if its version does not satisfy this requirement.

**Pull request merged**